src/components/panels/hours-panel-date-viewer.js

```diff
@@ -0,0 +1,586 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useStateValue } from '../use-state';
+
+const dateFormat = (string, abbreviated = false) => {
+  if (abbreviated) {
+    return string.toLocaleString('en-US', {
+      day: 'numeric',
+      month: 'short'
+    });
+  }
+  return string.toLocaleString('en-US', {
+    day: 'numeric',
+    month: 'long',
+    weekday: 'long',
+    year: 'numeric'
+  });
+};
+
+const IconWrapper = (props) => {
+  return (
+    <span
+      css={{
+        display: 'inline-block',
+        marginTop: '-2px'
+      }}
+      {...props}
+    />
+  );
+};
+
+const updateWeekOffset = ({
+  dispatch,
+  weekOffset,
+  event = null,
+  week = null,
+  relativeOffset = null
+}) => {
+  if (
+    !event
+    || event.type === 'click'
+    || (event.type === 'keydown' && (event.key === 'Enter' || event.key === ' '))
+  ) {
+    if (event) {
+      event.preventDefault();
+    }
+
+    let newOffset = weekOffset;
+
+    if (relativeOffset !== null) {
+      newOffset = weekOffset + relativeOffset;
+    } else if (week) {
+      const today = new Date();
+      const currentWeekStart = new Date(today.setDate(today.getDate() - today.getDay()));
+      const selectedWeekStart = new Date(week[0]);
+
+      newOffset = Math.round(
+        (selectedWeekStart.getTime() - currentWeekStart.getTime()) / (7 * 24 * 60 * 60 * 1000)
+      );
+    }
+
+    dispatch({
+      type: 'setWeekOffset',
+      weekOffset: newOffset
+    });
+  }
+};
+
+export default function HoursPanelDateViewer () {
+  const [isCalendarVisible, setIsCalendarVisible] = useState(false);
+  const [calendarStatus, setCalendarStatus] = useState('');
+
+  const handleKeydown = (event) => {
+    if (event.keyCode === 27) {
+      setIsCalendarVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeydown);
+
+    return () => {
+      document.addEventListener('keydown', handleKeydown);
+    };
+  });
+
+  const toggleCalendarVisibility = () => {
+    setIsCalendarVisible(!isCalendarVisible);
+    setCalendarStatus(
+      isCalendarVisible ? 'Calendar closed' : 'Calendar open'
+    );
+  };
+
+  return (
+    <div
+      css={{
+        background: 'var(--color-teal-100)',
+        borderBottom: '1px solid var(--color-neutral-100)',
+        borderTop: '1px solid var(--color-neutral-100)',
+        marginBottom: SPACING.L,
+        marginTop: SPACING.L,
+        padding: SPACING['2XS'],
+        position: 'sticky',
+        top: '-1px',
+        zIndex: 1
+      }}
+      id='dateViewerBar'
+    >
+      <div
+        aria-live='polite'
+        aria-atomic='true'
+        style={{
+          clip: 'rect(0 0 0 0)',
+          clipPath: 'inset(50%)',
+          height: '1px',
+          overflow: 'hidden',
+          position: 'absolute',
+          whiteSpace: 'nowrap',
+          width: '1px'
+        }}
```

Comment:
Is this to visually hide the element? If so, there is the `visually-hidden` class.
```css
.visually-hidden {
    position: absolute !important;
    height: 1px;
    width: 1px;
    overflow: hidden;
    clip: rect(1px 1px 1px 1px);
    clip: rect(1px, 1px, 1px, 1px);
}
```